test/readme.md

```diff
@@ -0,0 +1,23 @@
+# Test instruction
+
+For the testing tools, we use both nodeunit and [chai-passport-strategy](https://github.com/jaredhanson/chai-passport-strategy). Nodeunit is used for general testing purpose where the passport framework is not involved, and chai-passport-strategy is used for the passport strategy workflow testing. Instructions on how to use chai-passport-strategy can be found [here](https://github.com/jaredhanson/chai-passport-strategy/blob/master/README.md).
+
+All the test files should have a *_test suffix in their names and be placed in the correct subdirectory, depending on the testing tools used. The following is the rule: 
+
+* **nodeunit_test**: all the tests using nodeunit go here
+* **chai-passport_test**: all the tests using chai-passport-strategy go here
+* **resource**: all the resources needed for the tests (for example, pem key file) go here
+
+# How to run test on your machine
+
+In the passport-azure-ad root folder, type the following command in the command line to install the dependency packages:
+
+    $ npm install
+
+Then cd to the test folder, and type the following command:
```

Comment:
Also, is there a way to run tests without having to cd into the test folder? Can you use grunt to run_all_tests?